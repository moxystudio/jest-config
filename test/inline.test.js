'use strict';

const fs = require('fs');
const tmp = require('tmp-sync');
const inlineTransformer = require('../addons/with-web/inline');

const dir = tmp.in(__dirname);

afterAll(() => {
    tmp.clean();
});

it('should return inline content of file', () => {
    const fileContent = 'this is a file!';
    const filename = `${dir}/inline-test-file.txt`;

    fs.writeFileSync(filename, fileContent);

    const output = inlineTransformer.process(null, filename);

    expect(output).toContain(fileContent);
    expect(output).toMatchSnapshot();
});
