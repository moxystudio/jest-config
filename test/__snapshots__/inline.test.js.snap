// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return inline content of file 1`] = `"module.exports = \`this is a file!\`;"`;
